### PR TITLE
Improve auth alert dialog

### PR DESCRIPTION
**Pull request by NastuzziSamy**
68751023_377161493207391_5441813590841491456_n

69369475_2637724133121160_1823794090571988992_n

### CODE REVIEW

**Comment by aarrtthhuurr** [comment]
image

The pop-up just flashes on iOS and then I'm stuck on this

File: src/locales/en.json [hunk 1]
```diff
@@ -103,7 +103,11 @@
 			"login_placeholder": "CAS-UTC or email",
 			"password_label": "Password :",
 			"password_placeholder": "Connection password",
-			"bad_login_password": "Wrong identifiants",
+			"bad_login_password": "Wrong credentials",
+			"i_am_ext": "I am extern",
+			"i_am_cas": "I am CAS-UTC",
```
Comment (aarrtthhuurr) [inline src/locales/en.json]: "Try as an extern..."
"Try with CAS-UTC..."

File: src/locales/en.json [hunk 2]
```diff
@@ -103,7 +103,11 @@
...
+			"wrong_ext": "Check that you gave the right email adress you gave for your external contribution.",
+			"wrong_cas": "Check that you correctly wrote down your CAS-UTC login and password.",
```
Comment (aarrtthhuurr) [inline src/locales/en.json]: Only one simple message would be fine: "Check that you correctly wrote your credentials."

File: src/locales/fr.json [hunk 3]
```diff
@@ -106,6 +106,10 @@
 			"password_label": "Mot de passe :",
 			"password_placeholder": "Mot de passe de connexion",
 			"bad_login_password": "Mauvais identifiants",
+			"i_am_ext": "Je suis extérieur",
+			"i_am_cas": "Je suis CAS-UTC",
```
Comment (aarrtthhuurr) [inline src/locales/fr.json]: "Ressayer entant qu'externe..."
"Ressayer avec CAS-UTC..."

File: src/locales/fr.json [hunk 4]
```diff
@@ -106,6 +106,10 @@
...
+			"wrong_ext": "Vérifiez d'avoir bien écrit l'adresse email que vous avez fournie pour votre cotisation extérieure.",
+			"wrong_cas": "Vérifiez d'avoir bien écrit votre login CAS-UTC et d'avoir bien tapé votre mot de passe.",
```
Comment (aarrtthhuurr) [inline src/locales/fr.json]: "Vérifiez d'avoir bien écrit vos identifiants."

File: src/screens/Auth/AuthScreen.js [hunk 5]
```diff
@@ -28,6 +29,60 @@ class AuthScreen extends React.Component {
...
+	}
+
+	static openWrongExt() {
+		Alert.alert(
+			t('bad_login_password'),
+			t('wrong_ext'),
+			[
+				{
+					text: t('i_am_cas'),
+					style: Platform.OS === 'ios' ? 'destructive' : 'neutral',
```
Comment (aarrtthhuurr) [inline src/screens/Auth/AuthScreen.js]: Destructive is too much here. Neutral is fine.

File: src/screens/Auth/AuthScreen.js [hunk 6]
```diff
@@ -28,6 +29,60 @@ class AuthScreen extends React.Component {
...
+					onPress: () => AuthScreen.openWrongCas(),
+				},
+				...AuthScreen.getAlertButtons(),
+			],
+			{
+				cancelable: true,
```
Comment (aarrtthhuurr) [inline src/screens/Auth/AuthScreen.js]: iOS doesn't provide a "Cancel" option by default. You need to add a button with "cancel" style. (Usually placed in first according to RN documentation)

File: src/screens/Auth/AuthScreen.js [hunk 7]
```diff
@@ -28,6 +29,60 @@ class AuthScreen extends React.Component {
...
+			}
+		);
+	}
+
+	static openWrongCas() {
```
Comment (aarrtthhuurr) [inline src/screens/Auth/AuthScreen.js]: Same for here

File: src/locales/fr.json (hunk 3 above)
Comment (aarrtthhuurr) [reply src/locales/fr.json]: Or maybe juste "Essayer"

File: src/locales/en.json (hunk 1 above)
Comment (NastuzziSamy) [reply src/locales/en.json]: Actually, you can't try as something, the app auto detects what kind of type the user is. So he can be one of these types but not try as a type if it is not of the type

File: src/screens/Auth/AuthScreen.js (hunk 6 above)
Comment (NastuzziSamy) [reply src/screens/Auth/AuthScreen.js]: The option cancelable allows only the user to cancel the alert by clicking outside ^^

**Pull request merged**